torchvision/io/_video_opt.py

```diff
@@ -438,15 +447,20 @@ def _probe_video_from_memory(video_data):
     return info
 
 
-def _convert_to_sec(start_pts, end_pts, pts_unit, time_base):
+def _convert_to_sec(start_pts: float, end_pts: float, pts_unit: str, time_base: Fraction) -> Tuple[float, float, str]:
```

Comment:
It seems that `start_pts`, `end_pts`, `*pts*` switch between ints and floats in different methods. It's worth checking that in all signatures where we declare them as ints are indeed always ints. That might be easier to do with a debugger.

cc @prabhat00155

Reply:
As mentioned in the other comment, there is no clear dtype in https://github.com/pytorch/vision/blob/7947fc8fb38b1d3a2aca03f22a2e6a3caa63f2a0/torchvision/csrc/io/video_reader/video_reader.h#L29 and considering it's not casted afterwards, I'm not sure how to answer this :man_shrugging:

Reply:
Sorry I missed the other comment. I understand it's not straightforward to identify the types. That's why I think finding out what they are and annotating the code-base is useful. Nevertheless if we fail to annotate them correctly (say they are floats and we mark them as ints) it's going to be really confusing. 

Can we confirm the types of the various pts vars by running the unit-tests and putting a debugger to observe their materialized types?